Game Building crashes after update to 1.4.2

<!-- NOTE! The information you specify will be publicly accessible. -->
### Expected behaviour

### Actual behaviour

### Steps to reproduce

<hr/>

<table>
<tr><td>Build time</td><td>2023-01-31T13:11:13.422670</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>58bac6dd5a773e01f3a1816bde23283a38e5425d</td></tr>
<tr><td>Defold engine sha</td><td>8cd3a634b13f4db51a37607bf32bf3a3b362c8e6</td></tr>
<tr><td>Defold version</td><td>1.4.2</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"b9ce0926c26841b4b32dbc5d9ad52a4c"'>b9ce0926c26841b4b32dbc5d9ad52a4c</a></td></tr>
<tr><td>GPU</td><td>NVIDIA GeForce GTX 1650/PCIe/SSE2</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 NVIDIA 516.94</td></tr>
<tr><td>Java version</td><td>11.0.15+10-LTS</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 10</td></tr>
<tr><td>OS version</td><td>10.0</td></tr>
<table>

```
java.util.concurrent.FutureTask.report.class java.util.concurrent.ExecutionException: java.lang.AssertionError: Assert failed: (bt/content-hash? content-hash)
    at java.util.concurrent.FutureTask.report(FutureTask.java:122)
    at java.util.concurrent.FutureTask.get(FutureTask.java:191)
    at clojure.core$deref_future.invokeStatic(core.clj:2300)
    at clojure.core$future_call$reify.deref(core.clj:6974)
    at clojure.core$deref.invokeStatic(core.clj:2320)
    at clojure.core$deref.invoke(core.clj:2306)
    at clojure.core$map$fn.invoke(core.clj:2755)
    at clojure.lang.LazySeq.sval(LazySeq.java:42)
    at clojure.lang.LazySeq.seq(LazySeq.java:58)
    at clojure.lang.RT.seq(RT.java:531)
    at clojure.core$seq__5387.invokeStatic(core.clj:137)
    at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24)
    at clojure.core.protocols$fn.invokeStatic(protocols.clj:75)
    at clojure.core.protocols$fn.invoke(protocols.clj:75)
    at clojure.core.protocols$fn$G__8068__8086.invoke(protocols.clj:13)
    at clojure.core$reduce.invokeStatic(core.clj:6824)
    at editor.pipeline$batched_pmap.invokeStatic(pipeline.clj:174)
    at editor.pipeline$build_BANG_.invokeStatic(pipeline.clj:189)
    at editor.build$build_project_BANG_.invokeStatic(build.clj:59)
    at editor.app_view$build_project_BANG_.invokeStatic(app_view.clj:811)
    at editor.app_view$async_build_BANG_$phase_3_build_project_BANG___57449$run_project_build_on_background_thread_BANG___57450.invoke(app_view.clj:944)
    at editor.app_view$async_build_BANG_$run_on_background_thread_BANG___57429$fn.invoke(app_view.clj:871)
    at clojure.core$binding_conveyor_fn$fn.invoke(core.clj:2030)
    at clojure.lang.AFn.call(AFn.java:18)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:829)

editor.pipeline$make_dep_resources$fn__37722.invoke.class java.lang.AssertionError: Assert failed: (bt/content-hash? content-hash)
    at editor.pipeline$make_dep_resources$fn.invoke(pipeline.clj:128)
    at clojure.core$map$fn$fn.invoke(core.clj:2742)
    at clojure.core.protocols$fn.invokeStatic(protocols.clj:168)
    at clojure.core.protocols$fn.invoke(protocols.clj:124)
    at clojure.core.protocols$fn$G__8094__8108.invoke(protocols.clj:19)
    at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
    at clojure.core.protocols$fn.invokeStatic(protocols.clj:75)
    at clojure.core.protocols$fn.invoke(protocols.clj:75)
    at clojure.core.protocols$fn$G__8068__8086.invoke(protocols.clj:13)
    at clojure.core$transduce.invokeStatic(core.clj:6884)
    at clojure.core$into.invokeStatic(core.clj:6899)
    at editor.pipeline$make_dep_resources.invokeStatic(pipeline.clj:124)
    at editor.pipeline$build_BANG_$fn.invoke(pipeline.clj:201)
    at clojure.core$map$fn.invoke(core.clj:2755)
    at clojure.lang.LazySeq.sval(LazySeq.java:42)
    at clojure.lang.LazySeq.seq(LazySeq.java:51)
    at clojure.lang.Cons.next(Cons.java:39)
    at clojure.lang.RT.next(RT.java:709)
    at clojure.core$next__5371.invokeStatic(core.clj:64)
    at clojure.core$dorun.invokeStatic(core.clj:3142)
    at clojure.core$doall.invokeStatic(core.clj:3148)
    at editor.pipeline$batched_pmap$fn.invoke(pipeline.clj:176)
    at clojure.core$pmap$fn$fn.invoke(core.clj:7022)
    at clojure.core$binding_conveyor_fn$fn.invoke(core.clj:2030)
    at clojure.lang.AFn.call(AFn.java:18)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:829)
```

--- a/editor/src/clj/editor/code/script.clj
+++ b/editor/src/clj/editor/code/script.clj
@@ -32,6 +32,7 @@
             [editor.protobuf :as protobuf]
             [editor.resource :as resource]
             [editor.types :as t]
+            [editor.util :as eutil]
             [editor.validation :as validation]
             [editor.workspace :as workspace]
             [internal.util :as util]
@@ -524,37 +525,41 @@
               preprocessed-modules (lua-info->modules preprocessed-lua-info)
               proj-path->module-build-target (bt/make-proj-path->build-target module-build-targets)
               module->build-target (comp proj-path->module-build-target lua/lua-module->path)
-              preprocessed-module-build-targets (map module->build-target preprocessed-modules)
+              missing-modules (filterv (complement module->build-target) preprocessed-modules)]
+          (if (pos? (count missing-modules))
+            (g/->error _node-id :build-targets :fatal resource
+                       (str "Can't find required modules: " (eutil/join-words ", " " and " missing-modules)))
+            (let [preprocessed-module-build-targets (map module->build-target preprocessed-modules)
 
-              preprocessed-go-props-with-source-resources
-              (map (fn [{:keys [name type value]}]
-                     (let [go-prop-type (script-property-type->go-prop-type type)
-                           go-prop-value (properties/clj-value->go-prop-value go-prop-type value)]
-                       {:id name
-                        :type go-prop-type
-                        :value go-prop-value
-                        :clj-value value}))
-                   preprocessed-script-properties)
+                  preprocessed-go-props-with-source-resources
+                  (map (fn [{:keys [name type value]}]
+                         (let [go-prop-type (script-property-type->go-prop-type type)
+                               go-prop-value (properties/clj-value->go-prop-value go-prop-type value)]
+                           {:id name
+                            :type go-prop-type
+                            :value go-prop-value
+                            :clj-value value}))
+                       preprocessed-script-properties)
 
-              proj-path->resource-property-build-target
-              (bt/make-proj-path->build-target original-resource-property-build-targets)
+                  proj-path->resource-property-build-target
+                  (bt/make-proj-path->build-target original-resource-property-build-targets)
 
-              [preprocessed-go-props preprocessed-go-prop-dep-build-targets]
-              (properties/build-target-go-props proj-path->resource-property-build-target preprocessed-go-props-with-source-resources)]
-          ;; NOTE: The :user-data must not contain any overridden data. If it does,
-          ;; the build targets won't be fused and the script will be recompiled
-          ;; for every instance of the script component. The :go-props here describe
-          ;; the original property values from the script, never overridden values.
-          [(bt/with-content-hash
-             {:node-id _node-id
-              :resource (workspace/make-build-resource resource)
-              :build-fn build-script
-              :user-data {:lines preprocessed-lines
-                          :go-props preprocessed-go-props
-                          :modules preprocessed-modules
-                          :proj-path (resource/proj-path resource)}
-              :deps (into preprocessed-go-prop-dep-build-targets
-                          preprocessed-module-build-targets)})])))))
+                  [preprocessed-go-props preprocessed-go-prop-dep-build-targets]
+                  (properties/build-target-go-props proj-path->resource-property-build-target preprocessed-go-props-with-source-resources)]
+              ;; NOTE: The :user-data must not contain any overridden data. If it does,
+              ;; the build targets won't be fused and the script will be recompiled
+              ;; for every instance of the script component. The :go-props here describe
+              ;; the original property values from the script, never overridden values.
+              [(bt/with-content-hash
+                 {:node-id _node-id
+                  :resource (workspace/make-build-resource resource)
+                  :build-fn build-script
+                  :user-data {:lines preprocessed-lines
+                              :go-props preprocessed-go-props
+                              :modules preprocessed-modules
+                              :proj-path (resource/proj-path resource)}
+                  :deps (into preprocessed-go-prop-dep-build-targets
+                              preprocessed-module-build-targets)})])))))))
 
 (g/defnk produce-completions [completion-info module-completion-infos script-intelligence-completions]
   (code-completion/combine-completions completion-info module-completion-infos script-intelligence-completions))

